src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -519,6 +520,9 @@ private void copyDirectory(UnixPath source,
         try {
             mkdir(target, attrs.mode());
         } catch (UnixException x) {
+            if (x.errno() == EEXIST)
+                throw new FileSystemException(target.toString(), null,
+                    "Directory exists");
```

Comment:
I don't think "Directory exists", "Link exists", "Special file exists" will work for the exception message because the file that exists may be a regular file.

Reply:
Removed these messages in 171fbd201b77f8321040a5e81092bd5508a64b7f.